pwnlib/heap/glmalloc/process_informer.py

```diff
@@ -0,0 +1,51 @@
+import os.path
+import pwnlib.util.proc
+from pwnlib.heap.glmalloc.utils import \
+    u64, \
+    u32, \
+    get_libc_version_from_name, \
+    get_main_arena_addr
+
+
+class ProcessInformer:
+    """Helper class with information about process memory such as pointer_size,
+    and which allows perform operations over memory.
+
+    Attributes:
+        libc_version (tuple(int, int)): The glibc version in
+            format (major, minor)
+        pid (int): pid of the process
+        pointer_size (int): size in bytes of a pointer in the process
+        main_arena_address (int): address of the main arena malloc state
+
+    """
+
+    def __init__(self, pid, libc):
+        libc_name = os.path.basename(libc.path)
+        self.libc_version = get_libc_version_from_name(libc_name)
+        self.pid = pid
+
+        if "64" in libc.get_machine_arch():
+            self.pointer_size = 8
+            self.unpack_pointer = u64
+        else:
+            self.pointer_size = 4
+            self.unpack_pointer = u32
+
+        self.unpack_int = u32
+
+        self.main_arena_address = get_main_arena_addr(libc, self.pointer_size)
+
+    def read_memory(self, address, size):
+        with open('/proc/%s/mem' % self.pid, 'rb') as mem:
+            mem.seek(address)
+            return mem.read(size)
+
+    def maps(self):
+        return pwnlib.util.proc.MemoryMaps.from_process(self.pid)
```

Comment:
This should instead use `Corefile.mappings`

Reply:
done in c3d246a